### PR TITLE
Add popup

### PR DESCRIPTION
**Pull request by martinezpl**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "tslib": "^2.4.1",
     "typescript": "^4.9.3",
     "vite": "^4.0.0",
-    "webextension-polyfill": "^0.10.0"
+    "webextension-polyfill": "^0.10.0",
+    "pnpm": "^7.26.3"
```
Comment (a96lex) [inline package.json]: pnpm is not a dependency but a package manager. what we can do is add a link in readme to install it.

also, users could choose to use yarn, npm or smt else

File: src/app/stores/idStore.ts [hunk 2]
```diff
@@ -1,3 +1,10 @@
 import browser from "webextension-polyfill";
 
-export const owlieId: Promise<Record<string, any>> = (browser.storage.local.get("owlie-id") || browser.storage.sync.get("owlie-id"));
\ No newline at end of file
+export async function getId(): Promise<string> {
+    return (await (browser.storage.local.get("owlie-id") || browser.storage.sync.get("owlie-id")))["owlie-id"]
+}
+
+export async function setId(id: string): Promise<void> {
```
Comment (a96lex) [inline src/app/stores/idStore.ts]: ok this is nice. It should be shared.

please place this under `shared` directory. and pls run prettier

Comment (a96lex) [reply src/app/stores/idStore.ts]: hmm this is the only way we can communicate between apps. What i mean is that maybe is better to have a bit of abstraction. what do you think of renaming this functions and making them accept a `key` argument?

- `getFromLocalStorage(key: string)`
- `setToLocalStorage(key:string, value:string)`?

Comment (a96lex) [reply src/app/stores/idStore.ts]: so we can use to share some settings in the future

File: src/popup/Popup.svelte [hunk 3]
```diff
@@ -1 +1,73 @@
-<main>Popup placeholder</main>
+<script lang="ts">
```
Comment (a96lex) [inline src/popup/Popup.svelte:1]: run prettier on this file. and please lets keep structure the same across svelte files

- `<script>`
- `<html>`
- `<style>`

File: src/popup/Popup.svelte [hunk 4]
```diff
@@ -1 +1,73 @@
...
+    import {getId, setId} from "../app/stores/idStore"
+    let id = null;
+    let isLocked = true;
+  
+    function regenerateCode() {
```
Comment (a96lex) [inline src/popup/Popup.svelte]: what does this do? fetch localhost? pls add a comment somewhere to describe

File: src/popup/Popup.svelte [hunk 5]
```diff
@@ -1 +1,73 @@
...
+        color: rgb(0, 0, 0);
+      }
+  
+      input[type="text"] {
+        font-size: 1em;
+        width: 380px;
+      }
+    </style>
+    <div class="body">
+      <!-- svelte-ignore a11y-click-events-have-key-events -->
```
Comment (a96lex) [inline src/popup/Popup.svelte]: can we try to reduce svelte-ignore stuff? is it easy?

File: src/popup/Popup.svelte [hunk 6]
```diff
@@ -1 +1,73 @@
...
+      <b>Your code</b>
+      <input type="text" bind:value={id} readonly={isLocked} />
+      <small>
+        <!-- svelte-ignore a11y-invalid-attribute -->
+        <a class='regen' href="#" on:click={regenerateCode}>Regenerate code</a>
+      </small>
+      <br />
+      <br />
+      <p>You can copy paste your code into the donation message.<br/> Make sure it's set as <b>private</b> =)</p>
+      <iframe id='kofiframe' src='https://ko-fi.com/owliext/?hidefeed=true&widget=true&embed=true&preview=true' style='border:none;width:100%;padding:4px;background:#f9f9f9;' height='400' title='owliext'></iframe>
```
Comment (a96lex) [inline src/popup/Popup.svelte]: this inlyne style could be moved to `<style>` part

**Review by a96lex — COMMENTED** [review]
I added some comments, please review them

File: src/popup/Popup.svelte (hunk 4 above)
Comment (martinezpl) [reply src/popup/Popup.svelte]: it's just a placeholder for the endpoint regenerating the code

File: src/popup/Popup.svelte (hunk 5 above)
Comment (martinezpl) [reply src/popup/Popup.svelte]: it asks to add key events which doesn't make sense to me

**Pull request merged**